web/source/osk/bannerManager.ts

```diff
@@ -126,14 +165,30 @@ namespace com.keyman.osk {
         if(banner == this.activeBanner) {
           return;
         } else {
-          this.bannerContainer.removeChild(this.activeBanner.getDiv());
+          this.bannerContainer.replaceChild(banner.getDiv(), this.activeBanner.getDiv());
         }
       }
 
       this.activeBanner = banner;
       this.bannerContainer.appendChild(banner.getDiv());
     }
 
+    private invalidateSuggestions() {
```

Comment:
Neither this method nor `updateSuggestions` belongs in `ModelManager` - only in `SuggestionBanner`.